entity-framework/core/what-is-new/ef-core-7.0/whatsnew.md

```diff
@@ -3527,3 +3549,783 @@ Notice:
+
+<!--
+            #region CompositeIndex
+            modelBuilder
+                .Entity<Blog>()
+                .HasIndex(blog => new { blog.Name, blog.Owner })
+                .IsDescending(false, true);
+            #endregion
+-->
+[!code-csharp[CompositeIndex](../../../../samples/core/Miscellaneous/NewInEFCore7/ModelBuildingSample.cs?name=CompositeIndex)]
+
+Or, using a mapping attribute:
+
+<!--
+    [Index(nameof(Name), nameof(Owner), IsDescending = new[] { false, true })]
+    public class Blog
+    {
+        public int Id { get; set; }
+
+        [MaxLength(64)]
+        public string? Name { get; set; }
+
+        [MaxLength(64)]
+        public string? Owner { get; set; }
+
+        public List<Post> Posts { get; } = new();
+    }
+-->
+[!code-csharp[CompositeIndexByAttribute](../../../../samples/core/Miscellaneous/NewInEFCore7/ModelBuildingSample.cs?name=CompositeIndexByAttribute)]
+
+This results in the following SQL when using SQL Server:
+
+```sql
+CREATE INDEX [IX_Blogs_Name_Owner] ON [Blogs] ([Name], [Owner] DESC);
+```
+
+### Mapping attribute for composite keys
+
+EF7 introduces a new mapping attribute (aka "data annotation") for specifying the primary key property or properties of any entity type. Unlike <xref:System.ComponentModel.DataAnnotations.KeyAttribute?displayProperty=nameWithType>, [PrimaryKeyAttribute](https://github.com/dotnet/efcore/blob/main/src/EFCore.Abstractions/PrimaryKeyAttribute.cs) is placed on the entity type class rather than on the key property. For example:
+
+```csharp
+    [PrimaryKey(nameof(PostKey))]
+    public class Post
+    {
+        public int PostKey { get; set; }
+    }
+```
+
+This makes it a natural fit for defining composite keys:
+
+<!--
+    [PrimaryKey(nameof(PostId), nameof(CommentId))]
+    public class Comment
+    {
+        public int PostId { get; set; }
+        public int CommentId { get; set; }
+    }
+-->
+[!code-csharp[CompositePrimaryKey](../../../../samples/core/Miscellaneous/NewInEFCore7/ModelBuildingSample.cs?name=CompositePrimaryKey)]
+
+Defining the index on the class also means it can be used to specify private properties or fields as keys, even though these would usually be ignored when building the EF model. For example:
+
+```csharp
+[PrimaryKey(nameof(_id))]
+public class Tag
+{
+    private readonly int _id;
+}
+```
+
+### `DeleteBehavior` mapping attribute
+
+EF7 introduces a mapping attribute (aka "data annotation") to specify the <xref:Microsoft.EntityFrameworkCore.DeleteBehavior> for a relationship. For example, [required relationships](xref:core/modeling/relationships) are created with <xref:Microsoft.EntityFrameworkCore.DeleteBehavior.Cascade?displayProperty=nameWithType> by default. This can be changed to <xref:Microsoft.EntityFrameworkCore.DeleteBehavior.NoAction?displayProperty=nameWithType> by default using [DeleteBehaviorAttribute](https://github.com/dotnet/efcore/blob/main/src/EFCore.Abstractions/DeleteBehaviorAttribute.cs):
```

Comment:
```suggestion
EF7 introduces a mapping attribute (aka "data annotation") to specify the <xref:Microsoft.EntityFrameworkCore.DeleteBehavior> for a relationship. For example, [required relationships](xref:core/modeling/relationships) are created with <xref:Microsoft.EntityFrameworkCore.DeleteBehavior.Cascade?displayProperty=nameWithType> by default. This can be changed to <xref:Microsoft.EntityFrameworkCore.DeleteBehavior.NoAction?displayProperty=nameWithType> by default using [DeleteBehavior](https://github.com/dotnet/efcore/blob/main/src/EFCore.Abstractions/DeleteBehaviorAttribute.cs):
```